extras/find-diagnostic-tests.sh

```
@@ -0,0 +1,46 @@
+
+if [ $# -eq 0 ]; then
+  echo "Usage: $0 <error_code1> <error_code2> ..." >&2
+  echo "Example: $0 1 2 4 5 6 12 13 14 16 17 18 19 20 22 28 30 31" >&2
+  exit 1
+fi
+
+# Build regex pattern for error codes (matches "error N:" or "error N " where N is exact)
+codes_pattern=$(printf "|%s" "$@")
+codes_pattern="error (${codes_pattern:1})[^0-9]"
+
+echo "=== Searching for error codes: $@ ===" >&2
```

Shellcheck (0.11.0)

[error] Argument mixes string and array. Use * or separate argument.

(SC2145)